front/src/common/BootstrapSNCF/SelectSNCF.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { ReactNode } from 'react';
+import { isString } from 'lodash';
+import cx from 'classnames';
+
+export interface SelectOptionObject {
+  id?: string;
+  label: string;
+}
+interface SelectProps<T> {
+  id: string;
+  label?: ReactNode;
+  name?: string;
+  value?: T;
+  options: Array<T>;
+  onChange: (e?: T) => void;
+  className?: string;
+  sm?: boolean;
+}
+
+function SelectSNCF<T extends string | SelectOptionObject>({
+  id,
+  label,
+  name,
+  options,
+  onChange,
+  className,
+  sm,
+  value,
+}: SelectProps<T>) {
+  return (
+    <>
+      {label && <label htmlFor={id}>{label}</label>}
+      <select
+        id={id}
+        name={name}
+        onChange={(e) => {
+          const selected = e.target.value;
+          const item = options.find((o) => selected === (isString(o) ? o : o.id));
+          onChange(item);
+        }}
+        className={cx(className, sm && 'sm')}
+        value={isString(value) ? value : value?.id}
+      >
+        {options.map((option) => (
+          <option
+            key={isString(option) ? option : option.id}
+            value={isString(option) ? option : option.id}
+          >
+            {isString(option) ? option : option.label}
+          </option>
+        ))}
```

Comment:
It's only a suggestion, I find this clearer, but I'll let you decide what you prefer. You can resolve this comment on your own.

```suggestion
        {options.map((option) => {
          const optionId = isString(option) ? option : option.id;
          const optionLabel = isString(option) ? option : option.label;
          return (
            <option key={optionId} value={optionId}>
              {optionLabel}
            </option>
          );
        })}
```